tools/Dashboard/DevHome.Dashboard/Services/WidgetAdaptiveCardRenderingService.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveCards.Rendering.WinUI3;
+using DevHome.Common.Renderers;
+using DevHome.Common.Services;
+using DevHome.Contracts.Services;
+using DevHome.Dashboard.Helpers;
+using Microsoft.UI.Xaml;
+using Windows.Storage;
+using WinUIEx;
+
+namespace DevHome.Dashboard.Services;
+
+public class WidgetAdaptiveCardRenderingService : IAdaptiveCardRenderingService, IDisposable
+{
+    public event EventHandler RendererUpdated = (_, _) => { };
+
+    private readonly WindowEx _windowEx;
+
+    private readonly IThemeSelectorService _themeSelectorService;
+
+    private readonly SemaphoreSlim _rendererLock = new(1, 1);

```

Comment:
NIT: If using a `SemaphoreSlim` with a value of 1, convert it to a lock because both are equivalent.

Reply:
I don't think you can await inside a lock